arangod/Cache/FrequencyBuffer.h

```diff
@@ -154,12 +133,28 @@ class FrequencyBuffer {
   //////////////////////////////////////////////////////////////////////////////
   /// @brief Clear the buffer, removing all event records.
   //////////////////////////////////////////////////////////////////////////////
-  void clear() {
+  void clear() noexcept {
     for (std::size_t i = 0; i < _capacity; i++) {
       _buffer[i].store(T(), std::memory_order_relaxed);
     }
   }
+
+ private:
+  static constexpr std::size_t powerOf2(std::size_t capacity) {
```

Comment:
Marking the function as `constexpr` does not necessarily mean that is evaluated at compile time! It means that it _might_ be evaluated at compile time, given that all input parameters are known at compile time. However, in this case the function is called in the `FrequencyBuffer` construct with a capacity that is determined at runtime, so this function is also evaluated at runtime.

Reply:
Calculating `log2` and converting the result to an integer is very probably more expensive than the old version, but since the function is only called once during construction this is ok.